### PR TITLE
Implemented realsense2 FrameGrabber for depth, ir and color streams

### PR DESCRIPTION
**Pull request by cansik**
As discussed in #1306 I have implemented a first version of a realsense2 framegrabber. The API is a bit different to the original realsense camera, so I also had to change the API of FrameGrabber.

At the moment it is possible to enable as many streams as the user wants and also specifying custom streams. New frames are loaded through the `trigger()` method, so this method has to be called **before** `grab()`, `grabIR()`, `grabColor()` and so on.

Besides reading the different frames, it is also possible to list all the available cameras.

I have also included a simple example on how to use the depth stream and the mouse to see the distance of the camera at the mouse click position.

The code has been tested with a RealSense D435 on MacOS, but should run with every rs2 compatible device.

### CODE REVIEW

File: src/main/java/org/bytedeco/javacv/RealSense2FrameGrabber.java [hunk 1]
```diff
@@ -0,0 +1,504 @@
...
+    }
+
+    public RealSense2FrameGrabber(int deviceNumber) throws FrameGrabber.Exception {
+        this.deviceNumber = deviceNumber;
+
+        // create context
+        this.context = createContext();
+    }
+
+    public List<RealSense2DeviceInfo> getDeviceInfos() throws Exception {
```
Comment (saudet) [inline src/main/java/org/bytedeco/javacv/RealSense2FrameGrabber.java]: With this let's also implement
```java
public static String[] getDeviceDescriptions() throws FrameGrabber.Exception
```

**Comment by saudet** [comment]
`trigger()` doesn't seem to be flushing the frames, but anyway, would it be at least possible to make calls to that function optional? That way it wouldn't matter too much if it's not implemented correctly, for now.

File: src/main/java/org/bytedeco/javacv/RealSense2FrameGrabber.java (hunk 1 above)
Comment (cansik) [reply src/main/java/org/bytedeco/javacv/RealSense2FrameGrabber.java]: Ok, is implemented in the last commit.

**Comment by cansik** [comment]
Could you explain a bit further, what you mean with *flushing* the frames?

I could add the trigger code (`wait_for_frames`) into the `grab()` method, but this would lead to strange states, if `grab()` has not been called before another method like `grabIR()` is called.

The problem is that I have to `wait_for_frames` once per frameset. Otherwise, the different frames are not captured at the same time. So we need one method which is mandatory to be called before grabbing the frames. Of course this could be the `grab()` method itself.

**Comment by saudet** [comment]
What about calling wait_for_frames() on each grab(), unless trigger() has been called at least once?

Let's not worry about the flushing business for now. It will probably cause problems if someone tries to use it with, for example, a projector, but we can wait and see if that happens.

**Comment by cansik** [comment]
Yes that is possible. I will implement something like this. What is the boolean "triggerMode" used for? Could I use this as a flag to detect that or should I create a new one?

**Comment by saudet** [comment]
Yes, let's use that flag. If the user doesn't set it, then we shouldn't need to "trigger" it.

**Comment by cansik** [comment]
Ok, the last commit is using now the internal trigger mode :)

**Comment by cansik** [comment]
Is it possible that the travis job is stuck?

**Comment by cansik** [comment]
@saudet Anything else to do or is it ready to merge?

**Comment by saudet** [comment]
Should be good, yes. Thanks!

**Pull request merged**